hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -1626,6 +1638,22 @@ func (r *NodePoolReconciler) enqueueNodePoolsForConfig(obj client.Object) []reco
 		return enqueueParentNodePool(obj)
 	}
 
+	// Check if the ConfigMap is generated by an operator in the control plane namespace
+	// corresponding to this nodepool.
+	if _, ok := obj.GetLabels()[nodePoolOperatorCreatedConfigLabel]; ok {
+		// Go through the list of NodePools to determine which one
+		// this ConfigMap is referencing
+		for key := range nodePoolList.Items {
+			if obj.GetAnnotations()[nodePoolAnnotation] == nodePoolList.Items[key].Name &&
+				nodePoolList.Items[key].GetNamespace() == obj.GetNamespace() {
```

Comment:
`nodePoolList.Items[key].GetNamespace() == obj.GetNamespace()`
This will never be true, the generated config live in the control plane namespace. The NodePool live in a different namespace.

The control plane namespace is $hc.Namespace+$hc.Name (which is the same than hcp.Name), you can find the nodePool ns by dropping hcp.Name from the control plane ns, and use it to properly annotate the config .Annotations[nodePoolAnnotation] and then use enqueueParentNodePool.

Reply:
Thanks @enxebre. As discussed on the project-hypershift slack, I rewrote this to get the HostedControlPlane from the control plane namespace, and check it's hosted cluster (by annotation) to get the namespace of the NodePool.